### DISCUSSION TITLE
style will miss when use client route and webpack splitChunks

### DISCUSSION THREAD

**Discussion by okyping** [vercel/next.js#11869] [Help]
style will miss when use `Link` or `Router.push` and `webpack splitChunks`
only the script will be loaded, the style will not

**Comment by timneutkens** [comment]
It's hard to help based on this question, it does not indicate what your specific configuration is, are you using built-in css support? Are you customizing splitChunks? If so why are you customizing splitchunks?

**Comment by okyping** [comment]
eg：
next.config.js as below,

```
Object.assign(config.optimization, {
                usedExports: true,
                removeAvailableModules: true,
                mergeDuplicateChunks: true,
                minimize: true,
                checkWasmTypes: true,
                splitChunks: {
                    chunks: 'all',
                    minSize: 30000,
                    minChunks: 1,
                    maxAsyncRequests: 10,
                    maxInitialRequests: 10,
                    automaticNameDelimiter: '~',
                    name: true,
                    cacheGroups: {
                        react: {
                            test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
                            name: 'vendors-react',
                            chunks: 'all',
                            priority: 100
                        },
                        thirdDependents: {
                            test: /[\\/]node_modules[\\/](antd-mobile|next|next-server|axios|react-lazyload|clipboard|redux|react-redux|react-transition-group|url)[\\/]/,
                            name: 'vendors-dependents',
                            chunks: 'all',
                            priority: 90,
                        },
                        commons: {
                            test: /[\\/]node_modules[\\/]/,
                            name: 'vendors-commons',
                            chunks: 'all',
                            priority: 80,
                        },
                        util: {
                            test: /[\\/]util[\\/]/,
                            name: 'util',
                            chunks: 'all',
                            priority: 70,
                        },
                        components: {
                            test: /[\\/]components[\\/]/,
                            name: 'components',
                            chunks: 'all',
                            priority: 60,
                        },
                        styles: {
                            name: 'styles',
                            test: /\.css$/,
                            chunks: 'all',
                            enforce: true,
                            priority: 200
                        }
                    }
                }
 });
```
and use  Router.push to change router to search page
```
import Router from 'next/router';
...
enterSearchPage = e => {
       Router.push('/search', '/mt/search');
}
...
```

when router changed, just the js loaded, but not the css.

and the related code as below

```
if (!this.loadingRoutes[route]) {
        this.loadingRoutes[route] = true
        if (process.env.__NEXT_GRANULAR_CHUNKS) {
          this.getDependencies(route).then(deps => {
            deps.forEach(d => {
              if (
                /\.js$/.test(d) &&
                !document.querySelector(`script[src^="${d}"]`)
              ) {
                this.loadScript(d, route, false)
              }
              if (
                /\.css$/.test(d) &&
                !document.querySelector(`link[rel=stylesheet][href^="${d}"]`)
              ) {
                appendLink(d, 'stylesheet').catch(() => {
                  // FIXME: handle failure
                  // Right now, this is needed to prevent an unhandled rejection.
                })
              }
            })
            this.loadRoute(route)
          })
        } else {
          this.loadRoute(route)
        }
      }
```
the `process.env.__NEXT_GRANULAR_CHUNKS` will not be true，and then `appendLink` will not be executed

**Comment by jzwyz** [comment]
I ran into the same problem, but I used less

**Comment by ebs-PriyankaKanwar** [comment]
we have updated next JS, all things are working fine except one the CSS is not loaded in proper order in production mode, however, it's working fine in dev mode.

Here are the versions
Node: v13.7.0
React:^16.12.0
NextJs:9.1.6
OS: Linux - ubuntu

Here is what I got when did build:

├ static/pages/_app.js 16.1 kB
├ chunks/1899c3d08f1d4a9ab6e4e22050e66cd1600464e8.b9ea19.js 3.53 kB
├ chunks/48183758.ede116.js 65 B
├ chunks/57a5805a715b88beb8aec3ac43d8f962ef8879cc.53c4e4.js 2.34 kB
├ chunks/9ed67fcaced240be487e7a2880bfff868999eacb.76e83a.js 8.29 kB
├ chunks/ab68c4ca07cacf4e3048ee2d6f69810106b4b489.68a8b3.js 10.4 kB
├ chunks/d32b6132cd0996134c9f87f7274f553f791cd563.18dde5.js 6.39 kB
├ chunks/d6e4d54ae3ad4e326943499c7720eb830c61721b.b7693f.js 4.41 kB
├ chunks/framework.4dd100.js 40.3 kB
├ runtime/main.d3f0e8.js 13.2 kB
├ runtime/webpack.361a6a.js 1.59 kB
├ css/16c4518f007189356065.css 54.9 kB
└ css/f93669f8200e9712e425.css 30.6 kB

Here is my next.config :
const withFonts = require('next-fonts');
const withImages = require('next-images');
// const withCSS = require('@zeit/next-css')

const withPlugins = require('next-compose-plugins');
const nextEnv = require('next-env');
const dotenvLoad = require('dotenv-load');

dotenvLoad();

withFonts({
webpack(config, options) {
// config.plugins.push(new webpack.EnvironmentPlugin(localEnv))
return config;
}
})

module.exports = withPlugins([

nextEnv({
staticPrefix: 'CUSTOM_STATIC_',
publicPrefix: 'CUSTOM_PUBLIC_',
}),withFonts, withImages
]);

I have two things here
1 - why we are having two files of CSS when created build? How can we merge into one CSS file?
2 - How I can fix it so that it will work in the same manner as in dev mode?

Any suggestions will be helpful.
